src/esp32/hcsr04.rs

```diff
@@ -0,0 +1,289 @@
+// Support for HC-SR04 style ultrasonic ranging modules. See
+// https://cdn.sparkfun.com/datasheets/Sensors/Proximity/HCSR04.pdf.
```

Comment:
I will add comments here covering the configuration parameters with an example config for the sensor.